src/platform/packages/shared/kbn-esql-utils/src/utils/query_parsing_helpers.ts

```diff
@@ -591,3 +591,62 @@ export const missingSortBeforeLimit = (esql: string): boolean => {
   }
   return false;
 };
+
+/**
+ * Checks if the ESQL query contains a timeseries bucket aggregation.
+ * @param esql: string - The ESQL query string
+ * @param columns: DatatableColumn[] - The columns of the datatable
+ * @returns true if the query contains a timeseries bucket aggregation, false otherwise
+ */
+export function hasTimeseriesBucketAggregation(
+  esql: string,
+  columns: DatatableColumn[] = []
+): boolean {
+  const { root } = Parser.parse(esql);
+
+  const isTimeseriesCommand = Walker.commands(root).some(({ name }) => name === 'ts');
+  if (!isTimeseriesCommand) {
+    return false;
+  }
+
+  const columnsMap = new Map<string, DatatableColumn>();
+  columns.forEach((column) => {
+    const trimmedId = column.id.trim().replace(/\s+/g, '').toLowerCase();
+    columnsMap.set(trimmedId, column);
+  });
+
+  const statsCommands = Walker.commands(root).filter(({ name }) => name === 'stats');
+  if (statsCommands.length === 0) {
+    return false;
+  }
+
+  const byCommands = Walker.matchAll(statsCommands, { type: 'option', name: 'by' });
+
+  const lastByCommand = byCommands[Math.max(byCommands.length - 1, 0)];
+  const bucketColumns: string[] = [];
```

Comment:
Do you need the bucketColumns logic??

From the columns you can check if there is a date column, if there is no date column you can return. Then as long as you know that you have a date you check for timeseries and stats. 

If yes, then you check for bucket and tbucket breakdowns

The checks can be simple as the one I added above

```
  const tBucketFunctions = Walker.matchAll(lastByCommand, {
    type: 'function',
    name: 'tbucket',
  }) as ESQLFunction[];

  if (tBucketFunctions.length > 0) {
    return true;
  }

  const bucketFunctions = Walker.matchAll(lastByCommand, {
    type: 'function',
    name: 'bucket',
  }) as ESQLFunction[];

  if (bucketFunctions.length > 0) {
    return true;
  }
```

And that is all.